File: artemis-entity-tracker/src/main/java/net/namekdev/entity_tracker/connectors/DummyWorldUpdateListener.java
package net.namekdev.entity_tracker.connectors;

import com.artemis.utils.BitVector;
import net.namekdev.entity_tracker.model.ComponentTypeInfo;

public class DummyWorldUpdateListener implements WorldUpdateListener {
	@Override
	public void injectWorldController(WorldController controller) {
	}

	@Override
	public int getListeningBitset() {
		return 0;
	}

	@Override
	public void addedSystem(int index, String name, BitVector allTypes, BitVector oneTypes, BitVector notTypes) {}

	@Override
	public void addedManager(String name) {
	}

	@Override
	public void addedComponentType(int index, ComponentTypeInfo info) {
	}

	@Override
	public void updatedEntitySystem(int index, int entitiesCount, int maxEntitiesCount) {
	}

	@Override
	public void addedEntity(int entityId, BitVector components) {}

	@Override
	public void deletedEntity(int entityId) {
	}

	@Override
	public void updatedComponentState(int entityId, int componentIndex, Object valueTree) {
	}
}

File: artemis-entity-tracker/src/main/java/net/namekdev/entity_tracker/connectors/DummyWorldUpdateListener.kt
package net.namekdev.entity_tracker.connectors

import com.artemis.utils.BitVector
import net.namekdev.entity_tracker.model.ComponentTypeInfo

open abstract class DummyWorldUpdateListener : WorldUpdateListener {
    override fun injectWorldController(controller: WorldController) {}

    override val listeningBitset: Int
        get() = 0

    override fun addedSystem(index: Int, name: String, allTypes: BitVector, oneTypes: BitVector, notTypes: BitVector) {}

    override fun addedManager(name: String) {}

    override fun addedComponentType(index: Int, info: ComponentTypeInfo) {}

    override fun updatedEntitySystem(index: Int, entitiesCount: Int, maxEntitiesCount: Int) {}

    override fun addedEntity(entityId: Int, components: BitVector) {}

    override fun deletedEntity(entityId: Int) {}

    override fun updatedComponentState(entityId: Int, componentIndex: Int, valueTree: Any) {}
}
